cmd/entrypoint/runner.go

```diff
@@ -11,22 +13,52 @@ import (
 // stdout/stderr are collected -- needs e2e tests.
 
 // realRunner actually runs commands.
-type realRunner struct{}
+type realRunner struct {
+	signals chan os.Signal
+}
 
 var _ entrypoint.Runner = (*realRunner)(nil)
 
-func (*realRunner) Run(args ...string) error {
+func (rr *realRunner) Run(args ...string) error {
 	if len(args) == 0 {
 		return nil
 	}
 	name, args := args[0], args[1:]
 
+	// Receive system signals on "rr.signals"
+	if rr.signals == nil {
+		rr.signals = make(chan os.Signal, 1)
+	}
+	defer close(rr.signals)
+	signal.Notify(rr.signals)
+	defer signal.Reset()
+
 	cmd := exec.Command(name, args...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+	// dedicated PID group used to forward signals to
+	// main process and all children
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	// Start defined command
+	if err := cmd.Start(); err != nil {
+		return err
+	}
 
-	if err := cmd.Run(); err != nil {
+	// Goroutine for signals forwarding
+	go func() {
+		for s := range rr.signals {
+			// Forward signal to main process and all children
+			if s != syscall.SIGCHLD {
+				_ = syscall.Kill(-cmd.Process.Pid, s.(syscall.Signal))
```

Comment:
Curious - why do we negate cmd.Process.Pid here?

Reply:
If PID is negative, then the signal is sent to the process and all it's children. I will add a comment there to avoid any confusion.